### PR TITLE
Minor changes, mostly for aarch64 hyp

### PR DESCRIPTION
**Pull request by pingerino**

### CODE REVIEW

File: configs/seL4Config.cmake [hunk 1]
```diff
@@ -184,18 +184,25 @@ elseif(KernelArchArmV6)
 endif()
 if(KernelArmCortexA7)
     set(KernelArmCPU "cortex-a7" CACHE INTERNAL "")
+    config_set(KernelArmCacheLineSizeBits L1_CACHE_LINE_SIZE_BITS "6")
```
Comment (kent-mcleod) [inline configs/seL4Config.cmake]: This needs to go in src/arch/arm/config.cmake

Comment (kent-mcleod) [reply configs/seL4Config.cmake]: By needs I mean should.

Comment (pingerino) [reply configs/seL4Config.cmake]: on it

File: include/arch/arm/arch/64/mode/kernel/thread.h [hunk 2]
```diff
@@ -15,13 +15,25 @@ static inline word_t CONST sanitiseRegister(register_t reg, word_t v, bool_t arc
...
     if (reg == SPSR_EL1) {
         if (archInfo) {
-            switch (v & 0x1f) {
-            case PMODE_EL0t:
-            case PMODE_EL1t:
-            case PMODE_EL1h:
-                return v;
-            default:
-                break;
+            if (v & BIT(4)) {
```
Comment (kent-mcleod) [inline include/arch/arm/arch/64/mode/kernel/thread.h]: This commit probably needs to be reordered until after we support 32-bit guests on aarch64

Comment (pingerino) [reply include/arch/arm/arch/64/mode/kernel/thread.h]: removed this commit for now, I'll put it up later with other relevant changes

**Comment by pingerino** [comment]
Rebased onto master and removed commits that are no longer relevant

**Pull request merged**